### PR TITLE
Upgrade GitHub Actions

### PR DESCRIPTION
**Pull request by cclauss**
* https://www.pypy.org/download.html
* https://github.com/actions/checkout/releases
* https://github.com/actions/setup-python/blob/main/docs/advanced-usage.md#allow-pre-releases
* https://github.com/cygwin/cygwin-install-action/releases

### CODE REVIEW

**Comment by dnicolodi** [comment]
What did prompt these changes?

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -46,15 +46,15 @@ jobs:
           -
         include:
           - os: ubuntu
-            python: 'pypy-3.9'
+            python: 'pypy-3.10'
```
Comment (dnicolodi) [inline .github/workflows/tests.yml:49]: This does more than upgrading the github actions

Comment (cclauss) [reply .github/workflows/tests.yml:49]: Yes.  As mentioned in the commit message.

Comment (dnicolodi) [reply .github/workflows/tests.yml:49]: The commit message is literally "Upgrade GitHub Actions"

**Comment by rgommers** [comment]
> `python -m pip install mesonpy`

It's `meson-python` as the package name, not `mesonpy`.

**Comment by cclauss** [comment]
This seems to work...  (testing)
`python -m pip install -e 'git+https://github.com/mesonbuild/meson-python@main#egg=meson-python'`

**Comment by dnicolodi** [comment]
@rgommers Do you think that we can have the `MesonPy` project removed from PyPI? The last release is from 6 years ago and AFAICT it is not even installable with a modern `pip`.

**Comment by cclauss** [comment]
https://github.com/scikit-image/scikit-image/blob/80442f1cdb55e75ed4b8bb826e0632b4e1aea430/pyproject.toml#L116

**Comment by rgommers** [comment]
> @rgommers Do you think that we can have the `MesonPy` project removed from PyPI? The last release is from 6 years ago and AFAICT it is not even installable with a modern `pip`.

Good idea. I can give it a try - I've been through that process successfully before with a couple of other packages. 

> Oh. scikit-image seems to use it.

No, it's this:
```
requires = [
    'meson-python>=0.14',
```
The `build-backend` identifies the import name, not the package name.

**Comment by cclauss** [comment]
Thanks for the guidance!!  Now I see the error of my ways.

**Comment by rgommers** [comment]
I made a request to transfer the name on PyPI: https://github.com/Bartpab/meson-py/issues/6.

No worries at all @cclauss. 

I think we can still merge this, after undoing the `pypy-3.9` change. The rest of the changes seem fine as regular maintenance. WDYT @dnicolodi?

**Comment by rgommers** [comment]
> I made a request to transfer the name on PyPI: https://github.com/Bartpab/meson-py/issues/6.

This may take a while if we get no response: https://peps.python.org/pep-0541/#implementation.

**Comment by dnicolodi** [comment]
@rgommers Thanks. The confusion is compounded by the fact that some distributions decided to package `meson-python` as `python3-mesonpy`

**Comment by dnicolodi** [comment]
> I think we can still merge this, after undoing the `pypy-3.9` change. The rest of the changes seem fine as regular maintenance.

Sure. I'm just a bit wary of PRs touching project infrastructure from passers by.

**Pull request closed without merging**